cockatrice/src/remoteclient.h

```diff
@@ -85,28 +87,35 @@ private slots:
 private:
     static const int maxTimeout = 10;
     int timeRunning, lastDataReceived;
-
     QByteArray inputBuffer;
     bool messageInProgress;
     bool handshakeStarted;
-    bool newMissingFeatureFound(QString _serversMissingFeatures);
-    void clearNewClientFeatures();
+    bool usingWebSocket;
     int messageLength;
-
     QTimer *timer;
     QTcpSocket *socket;
+    QWebSocket *websocket;
     QString lastHostname;
     int lastPort;
-    QString getSrvClientID(const QString _hostname);
+
+    QString getSrvClientID(QString _hostname);
+    bool newMissingFeatureFound(QString _serversMissingFeatures);
+    void clearNewClientFeatures();
+    void connectToHost(const QString &hostname, unsigned int port);
+
 protected slots:
-    void sendCommandContainer(const CommandContainer &cont);
+    void sendCommandContainer(const CommandContainer &cont) override;
 
 public:
-    RemoteClient(QObject *parent = 0);
-    ~RemoteClient();
+    explicit RemoteClient(QObject *parent = nullptr);
+    ~RemoteClient() override;
     QString peerName() const
     {
-        return socket->peerName();
+        if (usingWebSocket) {
+            return socket->peerName();
```

Comment:
bug: inverted cases. If `usingWebSocket` you should return `websocket->peerName()`

Reply:
Good catch!